sdk/core/azure-core/src/main/java/com/azure/android/core/util/paging/PagedDataCollection.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package com.azure.android.core.util.paging;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+
+/**
+ * Represents a collection of paged data with fixed page size where pages can be retrieved synchrounously
+ * @param <T> type of the items contained in the page
+ * @param <P> page that contains the items
+ */
+public class PagedDataCollection<T, P extends Page<T>> {
```

Comment:
Not clear why there is a separate set of classes for paged data vs. paged data + response, but only for sync (for async there is only paged data + response).

Reply:
When autorest.android generates sync client, for each function in underlying service, e.g. service.getBlah(), sync client is supposed to have two versions of the same function in its interface, syncClient.getBlah(): Blah, and syncClient.getBlahWithResponse(): Response\<Blah\>. That's why we have two sets for sync client.

Reply:
For async client, the Callback\<Blah\> can return both Blah and okhttp3.Response. Therefore there is no need to have two function signatures for each service.getBlah function.

Reply:
Leaving as-is for this preview.